arm/cores/wiring_analog.c

```diff
@@ -68,41 +88,76 @@ XMC_VADC_GROUP_SetPowerMode( VADC_G0, XMC_VADC_GROUP_POWERMODE_NORMAL );
 XMC_VADC_GROUP_SetPowerMode( VADC_G1, XMC_VADC_GROUP_POWERMODE_NORMAL );
 
 #if(XMC_VADC_MAXIMUM_NUM_GROUPS > 2)
-/*Initialize Group*/
+/* Initialize Group */
 XMC_VADC_GROUP_Init( VADC_G2, &vadc_group_config );
+XMC_VADC_GROUP_Init( VADC_G3, &vadc_group_config );
 
 /* Switch on the converter of the Group*/
 XMC_VADC_GROUP_SetPowerMode( VADC_G2, XMC_VADC_GROUP_POWERMODE_NORMAL );
+XMC_VADC_GROUP_SetPowerMode( VADC_G3, XMC_VADC_GROUP_POWERMODE_NORMAL );
 #endif
 
 #endif
 /* Calibrate the VADC. Make sure you do this after all used VADC groups
-* are set to normal operation mode. */
+   are set to normal operation mode. */
 XMC_VADC_GLOBAL_StartupCalibration( VADC );
 
 /* Initialize the background source hardware. The gating mode is set to
-* ignore to pass external triggers unconditionally.*/
+   ignore to pass external triggers unconditionally.*/
 XMC_VADC_GLOBAL_BackgroundInit( VADC, &vadc_background_config );
 
+/* PC Mar-2019
+   Dummy read of ALL analogue inputs to ensure ALL analogue channels are
+   started in background scanning mode, otherwise first readings at least
+   will always be zero on reading an analogue input. */
+for( uint8_t chan = 0; chan < NUM_ANALOG_INPUTS; chan++ )
+   analogRead( chan );
+
 //Additional Initialization of DAC starting here
 }
 
 
-void analogReadResolution( uint8_t res )
+/* Set the resolution of analogRead return values in number of bits. 
+    Default is 10 bits (range from 0 to 1023)
+    Maximum is 12 bits (range from 0 to 4095)
+
+PC Sept-2019 Change flow to trap invalid input first and leave at old setting if
+    Invalid and return valid value or 255 for error, else return setting used
+     
+    Add set maximum value for resolution
+*/
+uint8_t analogReadResolution( uint8_t res )
 {
-if (res > ADC_MAX_READ_RESOLUTION)
-    res = ADC_MAX_READ_RESOLUTION;
+if( res > ADC_MAX_READ_RESOLUTION || res < ADC_MIN_RESOLUTION )
+  return 255;
+
 _readResolution = res;
+_readMaximum = (uint16_t)( ( (uint32_t)1U << res ) - 1 );
+return res;
 }
 
 
-void analogWriteResolution( uint8_t res )
+/* Set the resolution of analogWrite parameters in number of bits. 
+
+    Default (minimum) is 8 bits (range from 0 to 255).
+    Maximum is 16 bits (range 0 to 65535)
+
+PC Sept-2019 Change flow to trap invalid input first and leave at old setting if
+    Invalid and return valid value or 255 for error, else return setting used
+     
+    Add set maximum value for resolution
+  */
+uint8_t analogWriteResolution( uint8_t res )
```

Comment:
Arduino Reference Language compatibility needs to be ensured. This change breaks its. All API prototypes must be preserved.

Reply:
See previous comment this will not break anything as the situations where this could even **REMOTELY** break anything are not possible with Arduino licensing